pkg/virt-config/virt-config.go

```diff
@@ -36,7 +36,7 @@ const (
 	BandwithPerMigrationDefault                     = "0Mi"
 	MigrationAllowAutoConverge               bool   = false
 	MigrationAllowPostCopy                   bool   = false
-	MigrationProgressTimeout                 int64  = 150
+	MigrationProgressTimeout                 int64  = 300
```

Comment:
How do we determine sane value? 
Could we also make it more obvious that this is in seconds?

Reply:
The point of this timeout is to make sure that there is progress - that we do transfer any data.
This is mainly needed to avoid running into all kinds of very problematic corner cases. We would like to cancel the migration at the stage when we can still recover.
If no data is being transferred it would mean that QEMU buffer is full or it doesn't respond for another reason...

There are multiple examples, here is one: https://bugzilla.redhat.com/show_bug.cgi?id=1945532
The above could probably be resolved by the newly implemented `yank` command in qemu, but I'm not sure about the status of it.
However, there are many other problematic cases.

150 is just more than a TCP timeout - I could stick to it.
We also allow overriding this value in the KubeVirt CR, so I wouldn't change the default.

Reply:
@jean-edouard ^^^

Reply:
Doesn't libvirt already ensure that the connection is still alive? It also has its own timeout in seconds per GiB that we already leverage.
The two issues we have with `progressTimeout` right now are:
- It only resets when we see a remaining amount of data to transfer strictly inferior to the lowest value ever seen. This is a problem if for example 1GB of already-transferred memory gets modified, because the remaining amount of data to transfer will increase by that 1GB and won't go down below its lowest value ever for a while
- Custom debug prints showed that sometimes we just don't get any update from libvirt for a while, even if everything is fine. One test I did with and *idle* VM got no reported progress for 73 seconds
Thanks @vladikr for chiming in on this and giving it more context!

Reply:
> Doesn't libvirt already ensure that the connection is still alive? It also has its own timeout in seconds per GiB that we already leverage.

I'm not sure what are you referring to... We are using a direct connection between two qemu processes. libvirt queries the source QEMU to get the status of the migration. 
And then, here in the monitor we make decisions according to the status of the migration.

> The two issues we have with progressTimeout right now are:

> It only resets when we see a remaining amount of data to transfer strictly inferior to the lowest value ever seen. This is a problem if for example 1GB of already-transferred memory gets modified, because the remaining amount of data to transfer will increase by that 1GB and won't go down below its lowest value ever for a while

remainingData never increases, only the processedData which we don't use. 

> Custom debug prints showed that sometimes we just don't get any update from libvirt for a while, even if everything is fine. One test I did with and idle VM got no reported progress for 73 seconds

That may happen if the dirtyRate is very high, but still, this is less than our timeout...

What does `sometimes we just don't get any update from libvirt ` mean? We are calling getDomainStats every 400 mil. seconds, it should return something or error.

Thanks @vladikr for chiming in on this and giving it more context!

Reply:
Yes, will rollback. I confirm that `remainingData` does not increase. Thanks!

Reply:
Confirmed with more logs:
```
[...]
remainingData PROGRESS 3342336 TO 2818048. Disk: 2818048, Mem: 0
remainingData PROGRESS 2818048 TO 2424832. Disk: 2424832, Mem: 0
remainingData PROGRESS 2424832 TO 1966080. Disk: 1966080, Mem: 0
remainingData PROGRESS 1966080 TO 1638400. Disk: 1638400, Mem: 0
remainingData PROGRESS 1638400 TO 1179648. Disk: 1179648, Mem: 0
remainingData PROGRESS 1179648 TO 851968. Disk: 851968, Mem: 0
remainingData PROGRESS 851968 TO 65536. Disk: 65536, Mem: 0
remainingData NO UPDATE 65536 TO 1041833984. Disk: 0, Mem: 1041833984
remainingData NO UPDATE 65536 TO 1040658432. Disk: 0, Mem: 1040658432
remainingData NO UPDATE 65536 TO 1040211968. Disk: 0, Mem: 1040211968
remainingData NO UPDATE 65536 TO 1039785984. Disk: 0, Mem: 1039785984
remainingData NO UPDATE 65536 TO 1039360000. Disk: 0, Mem: 1039360000
remainingData NO UPDATE 65536 TO 1038934016. Disk: 0, Mem: 1038934016
remainingData NO UPDATE 65536 TO 1038508032. Disk: 0, Mem: 1038508032
[...]
```

Reply:
I haven't seen anything like this. We have had this exact mechanism in place for a while. 
If there was some kind of a recent change in the way `remainingData` is being calculated it would really be a good idea to track memory and disk separately.